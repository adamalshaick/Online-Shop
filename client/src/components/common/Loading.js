import React from "react";
import styled from "styled-components";

const Loader = styled.div`
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translateX(-50%);
  font-size: 2rem;
  font-weight: bold;
`;

const Dot = styled.span`
  position: absolute;
  left: 48%;
  top: 51%;
  transform: translate(-50%, -50%);
  font-size: 10rem;
  //   color: purple;
  opacity: 0;
  animation: sides 3s infinite;
  @keyframes sides {
    0% {
      opacity: 0;
    }

    25% {
      position: absolute;
      left: 50%;
      opacity: 1;
      font-size: 15rem;
    }

    50% {
      position: absolute;
      left: 52%;
      opacity: 0;
      font-size: 10rem;
    }

    75% {
      position: absolute;
      left: 50%;
      opacity: 1;
      font-size: 15rem;
    }

    100% {
      opacity: 0;
    }
  }
`;

const Loading = () => {
  return (
    <>
      <Loader>LOADING</Loader>
      <Dot>.</Dot>
    </>
  );
};

export default Loading;
