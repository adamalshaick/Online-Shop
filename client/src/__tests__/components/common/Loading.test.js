import React from "react";
import Loading from "../../../components/common/Loading";
import renderer from "react-test-renderer";

const wrapper = renderer.create(<Loading />).toJSON();

describe("render component", () => {
  it("renders component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
