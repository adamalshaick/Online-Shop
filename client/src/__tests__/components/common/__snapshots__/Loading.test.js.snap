// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
Array [
  <div
    className="sc-bdVaJa itEkXd"
  >
    LOADING
  </div>,
  <span
    className="sc-bwzfXH bWUJXc"
  >
    .
  </span>,
]
`;
